protocol_rfcs/iceberg-writer-compat-v1.md

```diff
@@ -0,0 +1,65 @@
+| [Default Columns](#default-columns)                                                               | No     | No                                         | N/A                                                                                                                                                                                                                                                      |
+| [Row Tracking](#row-tracking)                                                                     | No     | No                                         | N/A                                                                                                                                                                                                                                                      |
+| [Collations](https://github.com/delta-io/delta/blob/master/protocol_rfcs/collated-string-type.md) | No     | No                                         | N/A                                                                                                                                                                                                                                                      |
+| [Variant Types](#variant-data-type)                                                               | No     | No                                         | N/A                                                                                                                                                                                                                                                      |
+### Allowed Supported list of features
+To ensure that future features do not break tables with `IcebergWriterCompatV1` enabled, all enabled features must also be checked against an allowlist. Any enabled table features _must_ be in the list: [`appendOnly`, `columnMapping`, `icebergWriterCompatV1`, `icebergCompatV2`, `domainMetadata`, `vacuumProtocolCheck`, `v2Checkpoint`, `inCommitTimestamp`, `clustering`, `timestampNtz`, `typeWidening`]
+
+Additionally, the following features are allowed to be "supported", but must not be "active" (see [Disallowed Features](#disallowed-features)): [`invariants`, `changeDataFeed`, `checkConstraints`, `identityColumns`, `generatedColumns`]. These features, if supported, must be verified to be "inactive" via the checks specified above.
+
+We allow these legacy features to be "supported" because protocol updates can cause features to be carried over even though they are not in use. For example, if a table is on writer version 2, and then is updated to version 7, `invariants` can appear in the `writerFeatures` list because it was implicitly supported at version 2, even if it was not in use.
```

Comment:
maybe simplify this to
"We allow these legacy features to be "supported" because this table feature (which bumps the writer version to 7) enables legacy features automatically"